### PR TITLE
docs: Add documentation for event replay

### PR DESCRIPTION
**Pull request by iteriani**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/platform-browser/src/hydration.ts [hunk 1]
```diff
@@ -98,10 +98,23 @@ export function withI18nSupport(): HydrationFeature<HydrationFeatureKind.I18nSup
 }
 
 /**
- * Enables support for event replay
+ * Enables support for event replay when using clientside hydration.
+ * This feature scans all event listeners on the page and installs an event listener
+ * on the body before most JS has loaded and executed. Once this is done, the events
+ * are replayed so that relevant event listeners are executed.
```
Comment (AndrewKushnir) [inline packages/platform-browser/src/hydration.ts:104]: A proposal on how we can update docs to explain the feature (this info would be available on angular.dev), but don't describe implementation details:
```suggestion
 * Enables support for replaying user events (e.g. `click`s) that happened on a page
 * before hydration logic has completed. Once an application is hydrated, all captured events
 * are replayed and the relevant event listeners are executed.
```

File: packages/platform-browser/src/hydration.ts [hunk 2]
```diff
@@ -98,10 +98,23 @@ export function withI18nSupport(): HydrationFeature<HydrationFeatureKind.I18nSup
...
  *
+ * @usageNotes
+ *
+ * Basic example of how you can enable event replay in your application when
+ * `bootstrapApplication` function is used:
+ * ```
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideClientHyd()ration(withEventReplay)]
```
Comment (AndrewKushnir) [inline packages/platform-browser/src/hydration.ts:112]:
```suggestion
 *   providers: [provideClientHydration(withEventReplay())]
```

File: packages/platform-browser/src/hydration.ts [hunk 3]
```diff
@@ -100,8 +100,18 @@ export function withI18nSupport(): HydrationFeature<HydrationFeatureKind.I18nSup
 /**
  * Enables support for event replay
  *
+ * @usageNotes
+ *
+ * Basic example of how you can enable event replay in your application when
+ * `bootstrapApplication` function is used:
+ * ```
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideClientHydration(withEventReplay)]
```
Comment (eneajaho) [inline packages/platform-browser/src/hydration.ts:109]:
```suggestion
 *   providers: [provideClientHydration(withEventReplay())]
```

Shouldn't this be called?

**Review by AndrewKushnir — APPROVED** [review]

File: packages/platform-browser/src/hydration.ts (hunk 3 above)
Comment (AndrewKushnir) [reply packages/platform-browser/src/hydration.ts]: @eneajaho thanks for catching this, we've updated the docs.

**Comment by AndrewKushnir** [comment]
**Caretaker note**: docs-only changes, no presubmit required.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 0e23ed716ed99ff0ab607d370694fff85d1ee1e8.

**Pull request closed without merging**